### PR TITLE
DevTools: Download/Upload xml form data

### PR DESCRIPTION
**Pull request by bjosttveit**
## Description

This adds a button to download the form data as xml which can be used in prefill in localtest or you can directly upload (overwrite) the formdata of an existing instance using the upload button.

image

## Related Issue(s)
- https://github.com/Altinn/app-frontend-react/issues/1021#issuecomment-1478434144

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added/updated
  - [ ] Cypress E2E test(s) have been added/updated
  - [x] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [ ] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [x] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [x] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Changes/additions to component properties
  - [ ] Changes are reflected in both `src/layout/layout.d.ts` and `layout.schema.v1.json`, and these are all backwards-compatible
  - [x] No changes made
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [x] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

File: src/features/devtools/components/DownloadXMLButton/DownloadXMLButton.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        >
+          {({ getRootProps, getInputProps, isDragActive, isDragReject }) => (
+            <Button
+              {...getRootProps({
+                onClick: (e) => e.preventDefault(),
+              })}
+              variant='outline'
+              icon={<UploadIcon aria-hidden={true} />}
+            >
+              <input {...getInputProps()} />
```
Comment (Magnusrm) [inline src/features/devtools/components/DownloadXMLButton/DownloadXMLButton.tsx:66]: am not familiar with putting input fields inside of buttons. What is achieved by doing this and why do we need to do it this way? 😊

**Review by Magnusrm — APPROVED** [review]
Nice work! 👑 Just curious about one thing below.

File: src/features/devtools/components/DownloadXMLButton/DownloadXMLButton.tsx (hunk 1 above)
Comment (bjosttveit) [reply src/features/devtools/components/DownloadXMLButton/DownloadXMLButton.tsx:66]: Not familiar with this myself 😅 
But this is really a file uploader which is an `input`. And I am using the `DropZone` component like the one used in the file uploaders. This component requires an input element with the provided props to work with drag and drop. It seems to work as expected both when clicking and dropping files. And I am not too concerned about accessibility in the developer tools.

Comment (Magnusrm) [reply src/features/devtools/components/DownloadXMLButton/DownloadXMLButton.tsx:66]: Well, thats fine by me then 😁 Agree about the accessibility in DevTools 👍

**Pull request merged**